components/script/dom/node.rs

```diff
@@ -752,10 +752,9 @@ impl Node {
     }
 
     pub(crate) fn ranges_is_empty(&self) -> bool {
-        match self.rare_data().as_ref() {
-            Some(data) => data.ranges.is_empty(),
-            None => false,
-        }
+        self.rare_data()
+            .as_ref()
+            .is_none_or(|data| data.ranges.is_empty())
```

Comment:
I am fairly certain that this was previously wrong in the case where there is no rare data.